docs-src/concepts/what-is-versioning.md

```diff
@@ -0,0 +1,191 @@
+---
+id: what-is-versioning
+title: What is Versioning?
+sidebar_label: Versioning
+description: Versioning lets you more easily deploy changes to Workflow Definitions.
+tags:
+  - explanation
+---
+
+<!-- What is Versioning -->
+
+Versioning is a process allowing for the management and updating of Workflow code or Workflow Definitions without causing non-deterministic behavior in existing Workflows or Tasks.
+Workflow History contains the Execution History of a Workflow, including its Events, Tasks, Signals, and other information.
+When a Workflow Definition changes in an incompatible way, the new Workflow code can't be used to process the existing Workflow History, as the existing Workflow History and the new Workflow code have taken different code paths.
+Versioning allows Workflow to continue running on the old Workflow code, while new Workflows can start new code. This prevents Workflow Histories incompatibility issues.
+Using Versioning, you can deploy changes to your Workflows, as they evolve, without disrupting existing executions.
```

Comment:
```suggestion
Versioning is a collection of ways to deploy incompatible changes to Workflow code.
A code change is incompatible if a [Replay](/workflows#replays) of an existing Workflow Execution results in a [non-determinism error](/workflows#deterministic-constraints)—that is, if the code, given a past sequence of [Events](/workflows#event), takes a different code path and generates different [Commands](/workflows#command).
For more information, see [Code changes can cause non-deterministic behavior](/workflows#non-deterministic-change).

Versioning allows existing Workflows to continue running on the old Workflow code, while new Workflows can use the new code. 
This prevents non-determinism errors from occurring.
```

think this might be clearer